src/extension/index.ts

```diff
@@ -0,0 +1,239 @@
+import type { AppInfo } from "../appInfo.js"
+import type { CID } from "../common/cid.js"
+import type { DistinctivePath, Partition } from "../path/index.js"
+import type { Maybe } from "../common/types.js"
+import type { Permissions } from "../permissions.js"
+import type { Session } from "../session.js"
+import type { ShortHands } from "../index.js"
+
+import * as Events from "../events.js"
+import { VERSION } from "../index.js"
+
+
+// CREATE
+
+
+type Config = {
+  namespace: AppInfo | string
+  session: Maybe<Session>
+  capabilities?: Permissions
+  shorthands: ShortHands
+  lookupDataRoot: (username: string) => Promise<CID | null>
```

Comment:
Can we try to not use the shorthands from `index.ts` here?
You can use:
```ts
import { agent as agentDID } from "../did/index.js"
// Or:
import * as DID from "../did/index.js"
DID.agent(cryptoComponent)
```
In other places in the code we pass in the components entirely.
So I'd replace `shortHands` and `lookupDataRoot` with:
```ts
import { Crypto, Reference } from "../components.js"

// Following the pattern of other files
dependencies: { crypto: Crypto.Implementation; reference: Reference.Implementation }
```

Reply:
Yep, for sure! Updated to pass in the components directly instead of relying on shorthands and the `lookupDataRoot` function.